docs/reference/ccr/apis/follow/get-follow-stats.asciidoc

```diff
@@ -68,9 +68,12 @@ following tasks associated with each shard for the specified indices.
 [%collapsible%open]
 ====
 `fatal_exception`::
-(object) An object representing a fatal exception that cancelled the following
-task. In this situation, the following task must be resumed manually with the
-<<ccr-post-resume-follow,resume follower API>>.
+(object) An object representing a fatal exception that stopped the task in charge
+of replicating operations. In this situation, the following task must be paused manually
+with the <<ccr-post-pause-follow,pause follower API>> and then resumed with the
+<<ccr-post-resume-follow,resume follower API>>. If the index has been stopped by a fatal
```

Comment:
> Is the pause step necessary?

Yes it is: the persistent task framework does not have any concept of suspending and resuming a task. Here the pause step removes the shard follow persistent task and the resume step starts a new one.

> Also now that I think about it, shouldn't fatal exceptions normally be permanent, i.e. not something that a simple resume can fix? Maybe the real bug here is that we incorrectly see a fatal exception when the leader cluster is restarting?

In #92522 I saw that for some slow CI instances the follower cluster failed to connect to any of the leader cluster nodes and reports an IllegalStateException[^1]  which is considered as non retryable[^2]. I think that we should use a more appropriate exception instead of IllegalStateException for that case but I'm not confortable enough with the remote cluster logic to change that.

[^1]: see SniffConnectionStrategy.SniffClusterStateResponseHandler#handleNodes
[^2]: see ShardFollowNodeTask#shouldRetry

Reply:
Ugh yeah that's wrong, I thought we used `NoSeedNodeLeftException` (which is considered retryable) for that case but apparently not.

Reply:
However we do get `NoSeedNodeLeftException` if we just cannot talk to the remote cluster. What I think might be happening is that we get a list of nodes from the remote cluster just before it shuts down, which means we then fail to connect to any of them.

Reply:
Yes, that's probably what is happening. 

Following what you said, we should change the `IllegalStateException` to become a retryable `NoSeedNodeLeftException` and revert what I did in #92522 to not pause/resume (or recreate) since the retries should force the remote cluster reconnection.

For the documentation change, advising to simply resume on fatal exceptions does not work I think without pausing first. We can advise to pause, or we can maybe just link to the `Recreate a follower index` paragraph (which documents how to pause, unfollow, close and follow again the index.